### PR TITLE
Add `ValueSum::commit`

### PR DESCRIPTION
**Pull request by jarys**
This PR adds API for creating `ValueCommitment`s.

It enables us serialize less data when passing proof witness from HWW to a Prover.

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -174,6 +174,15 @@ impl ValueSum {
             sign,
         )
     }
+
+    /// Commits this value with the corresponding trapdoor `rcv` by $ValueCommit^Orchard$
```
Comment (daira) [inline src/value.rs:178]:
```suggestion
    /// Commits this value with the corresponding trapdoor `rcv` by ValueCommit^Orchard.
```
(LaTeX would not render correctly here.)

File: src/value.rs [hunk 2]
```diff
@@ -174,6 +174,15 @@ impl ValueSum {
...
+    ///
+    /// Defined in [Zcash Protocol Spec § 5.4.8.3: Homomorphic Pedersen commitments (Sapling and Orchard)][concretehomomorphiccommit].
+    ///
+    /// [concretehomomorphiccommit]: https://zips.z.cash/protocol/nu5.pdf#concretehomomorphiccommit
+    pub fn commit(self, rcv: ValueCommitTrapdoor) -> ValueCommitment {
+        ValueCommitment::derive(self, rcv)
```
Comment (daira) [inline src/value.rs:184]: Why not make `ValueCommitment::derive` public instead?

**Review by daira — COMMENTED** [review]
I'm not sure it's necessary to add a new method on `ValueSum`, as opposed to making the existing `ValueCommitment::derive` method public.

**Comment by jarys** [comment]
Yeah, we can just publicize `ValueCommitment::derive`.

Closing this PR in favor of #351.

**Pull request closed without merging**